.github/workflows/publish.yml

```diff
@@ -20,6 +20,10 @@ jobs:
   # calls our general CI workflow (tests, build docs, etc.)
   tests:
     uses: ./.github/workflows/CI.yml
+    # needed for the coverage action
+    permissions:
+      contents: write
+      pull-requests: write
```

Comment:
I'm slightly confused why this is needed even though the `coverage` job has a `if: github.event.workflow_run.event != 'release'` clause. But as you say, permissions are weird.  I think you were saying it's because coverage runs anyway (for some reason) and pushes the coverage data to a new branch?  Would be nice to just make it so that *coverage doesn't try to run at all* on release workflows... but I'm assuming that since you're not doing that here, it's probably not that simple. 🤷🏻

Reply:
It all seems to come from the fact that we have the coverage step within our main CI workflow, which is in turn called by the release workflow. 
So even if the coverage comment part is not triggered per se, GH actions checks that the permissions granted within the caller workflow (release) and the called one (CI) match. 

The one way I could think of decoupling these workflows would be to split the tests and docs building jobs into a reusable action and call that within the CI and release workflows independently but that seemed like too much hassle. 
I can do that if you see it as worthwhile.